File: checkers/python/weak-random.test.py
import random
import secrets

# Vulnerable: random.random() for token
# <expect-error>
token = str(random.random())

# Vulnerable: random.choices for password
# <expect-error>
password = ''.join(random.choices('abc123', k=16))

# Vulnerable: random.randint for security
# <expect-error>
code = random.randint(100000, 999999)

# Vulnerable: from random import
# <expect-error>
from random import choice, randrange

# <no-error> - Using secrets module
secure_token = secrets.token_hex(32)

# <no-error> - secrets.randbelow
secure_int = secrets.randbelow(1000)

# <no-error> - secrets.choice
secure_choice = secrets.choice(['a', 'b', 'c'])

# <no-error> - os.urandom
import os
key = os.urandom(32)

File: checkers/php/weak_random.test.php
<?php
// Vulnerable: rand() for token generation
// <expect-error>
$token = rand();

// Vulnerable: mt_rand() for security
// <expect-error>
$sessionId = md5(mt_rand());

// Vulnerable: uniqid() for token
// <expect-error>
$resetToken = uniqid();

// Vulnerable: array_rand for password
$chars = str_split('abc123');
// <expect-error>
$char = $chars[array_rand($chars)];

// <no-error> - Using random_bytes
$secureToken = bin2hex(random_bytes(32));

// <no-error> - Using random_int
$secureNumber = random_int(100000, 999999);

// <no-error> - Static rand for non-security use
// Note: This would still flag, but in practice
// rand() should only be used for non-security
?>
